crates/polars-stream/src/physical_plan/lower_expr.rs

```diff
@@ -1135,6 +1135,92 @@ fn lower_exprs_with_ctx(
                 transformed_exprs.push(ctx.expr_arena.add(AExpr::Column(value_key.clone())));
             },
 
+            // pl.row_index() maps to this.
```

Comment:
Note: this had quietly regressed back to the old state since #23633 because `int_range` is marked as row separable, and the lowering of that was done in a match arm above.